cap-primitives/src/std/fs/open.rs

```diff
@@ -53,6 +53,12 @@ pub fn open(start: &fs::File, path: &Path, options: &OpenOptions) -> io::Result<
             Err(result_error) => match result_error.kind() {
                 io::ErrorKind::PermissionDenied | io::ErrorKind::InvalidInput => (),
                 _ => {
+                    let unchecked_error = match unchecked_error {
+                        OpenUncheckedError::Io(err) => err,
+                        OpenUncheckedError::Symlink => {
+                            io::Error::new(io::ErrorKind::Other, "unexpected ELOOP or EMLINK")
```

Comment:
Agreed, this sounds like a good idea. I've done it now in 30e4ce5. We could in theory refactor `OpenUncheckedError` into a struct with some `kind` enum since currently both variants store `io::Error`, but I think it's OK as it is now since the error enum is so simple and not used in too many places.